modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,34 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones || nil
```

Comment:
Can you move this into the constructor? No need to set it to nil, nil is returned for missing settings.